Sources/TuistCore/MetadataProviders/XCFrameworkSignatureProvider.swift

```diff
@@ -0,0 +1,104 @@
+            case (.some, nil):
+                return "The framework at \(path) signed with an Apple certificate lacks the team name."
+            default:
+                return "The framework at \(path) signed with an Apple certificate lacks the team identifier and name."
+            }
+        }
+    }
+}
+
+/// A provider that can verify the signature calculated from an XCFramework.
+public struct XCFrameworkSignatureProvider {
+    private let commandRunner: CommandRunning
+    private let fileSystem: FileSysteming
+    private let codesignController: CodesignControlling
+
+    public init(
+        commandRunner: CommandRunning = CommandRunner(),
+        fileSystem: FileSysteming = FileSystem(),
+        codesignController: CodesignControlling = CodesignController()
+    ) {
+        self.commandRunner = commandRunner
+        self.fileSystem = fileSystem
+        self.codesignController = codesignController
+    }
+
+    private static let signedWithAppleCertificateString = "Authority=Apple Root CA"
+    private static let teamNameRegExPattern = #"Authority=[^:]+?:\s*([^()]+)\s*\(([A-Z0-9]+)\)"#
+    private static let teamIdentifierRegExPattern = #"TeamIdentifier=([A-Z0-9]+)"#
+
+    /// Returns the signature of the XCFramework at the given `xcframeworkPath`.
+    public func signature(of xcframeworkPath: Path.AbsolutePath) async throws -> XCFrameworkSignature {
+        guard let output = try await codesignController.codesignSignature(of: xcframeworkPath) else {
+            return .unsigned
+        }
+
+        guard output.contains(Self.signedWithAppleCertificateString) else {
+            let fingerprint = try await extractFingerprint(from: xcframeworkPath)
+            return .selfSigned(fingerprint: fingerprint)
+        }
+
+        let teamIdentifier = (try? RegEx(pattern: Self.teamIdentifierRegExPattern).matchGroups(in: output).first?.first)?
+            .trimmingCharacters(in: .whitespaces)
+        let teamName = (try? RegEx(pattern: Self.teamNameRegExPattern).matchGroups(in: output).first?.first)?
+            .trimmingCharacters(in: .whitespaces)
+
+        guard let teamIdentifier, let teamName else {
+            throw XCFrameworkSignatureProviderError.appleCertificateSignedXCFrameworkMissingDetails(
+                xcframeworkPath,
+                teamIdentifier: teamIdentifier,
+                teamName: teamName
+            )
+        }
+
+        return .signedWithAppleCertificate(teamIdentifier: teamIdentifier, teamName: teamName)
+    }
+
+    private func extractFingerprint(from xcframeworkPath: Path.AbsolutePath) async throws -> String {
+        try await fileSystem.runInTemporaryDirectory(prefix: "xcframework-signature-extractor)") { temporaryPath in
+            try await codesignController.codesignExtractSignature(of: xcframeworkPath, into: temporaryPath)
+
+            let certFile: Path.AbsolutePath = temporaryPath.appending(component: "codesign0")
```

Comment:
```suggestion
            let certFile = temporaryPath.appending(component: "codesign0")
```

nit: I don't think the explicit type should be needed here